// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

enum BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnum {
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumNO_VALUE_DO_NOT_USE = 0;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumSIGNATURE_ALGORITHM_UNSPECIFIED = 1;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumRSA_PSS_2048_SHA256 = 2;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumRSA_PSS_3072_SHA256 = 3;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumRSA_PSS_4096_SHA256 = 4;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumRSA_PSS_4096_SHA512 = 5;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumRSA_SIGN_PKCS1_2048_SHA256 = 6;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumRSA_SIGN_PKCS1_3072_SHA256 = 7;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumRSA_SIGN_PKCS1_4096_SHA256 = 8;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumRSA_SIGN_PKCS1_4096_SHA512 = 9;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumECDSA_P256_SHA256 = 10;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumEC_SIGN_P256_SHA256 = 11;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumECDSA_P384_SHA384 = 12;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumEC_SIGN_P384_SHA384 = 13;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumECDSA_P521_SHA512 = 14;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnumEC_SIGN_P521_SHA512 = 15;
}

message BinaryauthorizationAttestor {
  string name = 1;
  string description = 2;
  BinaryauthorizationAttestorUserOwnedGrafeasNote user_owned_grafeas_note = 3;
  string update_time = 4;
  string project = 5;
}

message BinaryauthorizationAttestorUserOwnedGrafeasNote {
  string note_reference = 1;
  repeated BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeys public_keys = 2;
  string delegation_service_account_email = 3;
}

message BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeys {
  string comment = 1;
  string id = 2;
  string ascii_armored_pgp_public_key = 3;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKey pkix_public_key = 4;
}

message BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKey {
  string public_key_pem = 1;
  BinaryauthorizationAttestorUserOwnedGrafeasNotePublicKeysPkixPublicKeySignatureAlgorithmEnum signature_algorithm = 2;
}

message ApplyBinaryauthorizationAttestorRequest {
  BinaryauthorizationAttestor resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteBinaryauthorizationAttestorRequest {
  string service_account_file = 1;
  BinaryauthorizationAttestor resource = 2;
}

message ListBinaryauthorizationAttestorRequest {
  string service_account_file = 1;
  string Project = 2;
}

message ListBinaryauthorizationAttestorResponse {
  repeated BinaryauthorizationAttestor items = 1;
}

service BinaryauthorizationAttestorService {
  rpc ApplyBinaryauthorizationAttestor(ApplyBinaryauthorizationAttestorRequest) returns (BinaryauthorizationAttestor);
  rpc DeleteBinaryauthorizationAttestor(DeleteBinaryauthorizationAttestorRequest) returns (google.protobuf.Empty);
  rpc ListBinaryauthorizationAttestor(ListBinaryauthorizationAttestorRequest) returns (ListBinaryauthorizationAttestorResponse);
}
